src/test/java/server/RequestParserTest.java

```diff
@@ -0,0 +1,39 @@
+package server;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class RequestParserTest {
+
+    @Test
+    void itParsesARequestWithNoBody() throws IOException {
```

Comment:
**praise**: I like that you included an integration test here!

I also appreciate the abstraction level. You are making an assertion about an **HTTP request** vs an attribute of the request, which resonates for me when we are thinking about the parser as a whole.